soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -192,11 +193,55 @@ def main() -> None:
         padding_len = periphconf_size - len(periphconf_combined)
         periphconf_final = periphconf_combined + bytes([0xFF for _ in range(padding_len)])
 
-        if kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF") == "y":
+        # Configure PERIPHCONF field based on whether this is a recovery build
+        is_recovery_build = kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF_RECOVERY") == "y"
+        is_normal_periphconf = kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF") == "y"
+
+        if is_recovery_build:
+            # Recovery build: populate PERIPHCONF_RECOVERY field
+            uicr.PERIPHCONF_RECOVERY.ENABLE = ENABLED_VALUE
+            uicr.PERIPHCONF_RECOVERY.ADDRESS = periphconf_address
+            uicr.PERIPHCONF_RECOVERY.MAXCOUNT = math.floor(periphconf_size / 8)
+        elif is_normal_periphconf:
+            # Normal build: populate PERIPHCONF field
```

Comment:
Don't we need both of these to be populated? We only have one UICR being generated AFAIK, so in a "complete" build I would think that both of these need to be set